### ISSUE TITLE
PR/issue labeling bot isn't labeling

### ISSUE THREAD

**Issue by stephentoub** [dotnet/runtime#82643]
Nothing is getting automatically labeled by area, for at least the last day or two.

**Comment by ericstj** [comment]
@dakersnar can you have a look and see if something stopped?

**Comment by ericstj** [comment]
I'm trying to find the App Services mentioned here https://github.com/dotnet/issue-labeler/tree/main/Documentation#web-app-service-list and don't see them.  So either I don't have access -- or they've been removed and that's why our labeler stopped.

**Comment by ericstj** [comment]
Ok, I was able to determine what to do to poke the service.  I can run https://dotnet-runtime-issue-labeler.azurewebsites.net/api/WebhookIssue/load/dotnet/runtime to make the service to load the dotnet/runtime model.  I can also run https://dotnet-runtime-issue-labeler.azurewebsites.net/api/WebhookIssue/dotnet/runtime/82648 to test that it's able to label a recent PR.  So the problem seems to be the connection between the repo events and the labeler.

**Comment by ericstj** [comment]
The mystery I need to figure out is this TODO: https://github.com/dotnet/issue-labeler/tree/main/Documentation#setting-up-github-webhooks-for-fully-automated-labeling

🤔

**Comment by stephentoub** [comment]
Thanks for investigating! That is an amusing and frustrating to do :-)

**Comment by ericstj** [comment]
Could be related to this: https://github.com/dotnet/issue-labeler/pull/53 however the timing doesn't really line up.

**Comment by stephentoub** [comment]
FWIW, i didn't dig back to see how long it's been down, just that's its been at least a day.

**Comment by lewing** [comment]
I've noticed missed area-codeflow labels for several days, that could be it

**Comment by ericstj** [comment]
Probably it can wait a bit longer.  Might need to get some help from @jeffhandley and @maryamariyan.  I still can't quite figure out the piece that connects the new events to the service.  I think it has to do with https://github.com/apps/dotnet-issue-labeler but I'm not certain.

FWIW I noticed that the labels seemed to stop happening on 2/16 with agrees with this being related to @jeffhandley's change.

**Comment by jeffhandley** [comment]
Shoot; I'm sorry. I tested as much as I could figure out how to (both locally and in the deployment). I also saw manually triggered events working; I thought I saw some automated events flow through and work as expected too.

I can work on it Wednesday and I'll make sure all issues are caught up.

**Comment by jaredpar** [comment]
This is failing for dotnet/roslyn too now.

**Comment by dakersnar** [comment]
I'm attempting to deploy a temporary fix now...

**Comment by dakersnar** [comment]
Fixed: https://github.com/dotnet/runtime/issues/82721. @jaredpar if you want to test dotnet/roslyn, feel free.

**Comment by dakersnar** [comment]
@jeffhandley I (locally) reverted your 2/16 change and redeployed targeting net5.0 instead of net7.0. One of those two variables caused the problem, so feel free to add them back one at a time when you get back and see what happens. I'm going to leave it in a working state for now.

**Comment by jaredpar** [comment]
How did you fix it? Like @ericstj I can't access / see any of the subs listed in the documentation.

**Comment by jaredpar** [comment]
@dakersnar whatever you did worked. Thanks! 

Can you tell us what you did?

**Comment by dakersnar** [comment]
@jaredpar I summarized it above (not sure if you saw that), but to add more details, the documentation for deploying the bot can be found here: https://github.com/dotnet/issue-labeler/tree/main/Documentation. Specifically, the `Publish the labeler app from Visual Studio` section.

On my local clone of `issue-labeler`, on the `feature/public-dispatcher` branch, I used `git checkout` to go back to the last commit before the 2/16 changes and followed those steps. Notably, when Jeff and I were deploying the changes on 2/16, we tried to target .net7.0 instead of .net5.0. That theoretically could have caused this problem, so I also went back to targeting net5.0.

In conclusion, either Jeff's 2/16 changes or the upgrade to net7.0 caused the break. We likely want to look into both of those possibilities.

**Comment by ericstj** [comment]
Another possibility is simply that there was something wrong about how it was deployed with Jeff's changes (and the changes themself were OK).  I still don't understand where the piece that listens to GitHub and applies changes to PRs and issues is - can you point me to that?

**Comment by ericstj** [comment]
I think I found it.  I was looking in `main` instead of `feature/public-dispatcher` - which seems to be the current production branch.  It looks like a post to this endpoint is what triggers the issue update.  https://github.com/dotnet/issue-labeler/blob/0451a627ba0415cdf89de9097aa522c3e6c0ff94/src/Microsoft.DotNet.GitHub.IssueLabeler/Controllers/WebhookIssueController.cs#L53.  It looks to me like a [webhook]( https://docs.github.com/webhooks-and-events/webhooks/webhook-events-and-payloads) must be configured to post to this endpoint for PR and issue creation.

I was testing by using the get endpoints, not the post endpoint.  The post endpoint is the only thing that actually interacts with GitHub to update the issue/PR.

**Comment by ViktorHofer** [comment]
Can this issue be closed now?

**Comment by stephentoub** [comment]
Yup
